TabNine.py

```diff
@@ -361,11 +367,40 @@ def on_selection_modified_async(self, view):
 
         if self.choices == []:
             view.hide_popup()
+            self.clear_delay_timer()
         else:
-            my_show_popup(view, to_show, substitute_begin)
-            self.popup_is_ours = True
-            self.seen_changes = False
+            if view.is_popup_visible():
+                self.show_competion_dialog(view, to_show, substitute_begin)
+            else:  
```

Comment:
below is the main change:
it will clear the timeout if any and set a new one that will open the completion popup after the`auto_complete_delay` if it has a value greater or equal to 1000ms